shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -351,6 +351,13 @@ - (bool)runWithEntrypoint:(NSString*)entrypoint libraryURI:(NSString*)libraryURI
                    << entrypoint.UTF8String;
   } else {
     [self maybeSetupPlatformViewChannels];
+  }
+
+  return _shell != nullptr;
+}
+
+- (bool)runWithEntrypoint:(NSString*)entrypoint libraryURI:(NSString*)libraryURI {
+  if ([self createShell:entrypoint libraryURI:libraryURI]) {
```

Comment:
What happens if someone calls `launchEngine` with a different set of arguments than what was passed to an earlier call to `createShell`?  Does it make sense to provide an API that allows them to be different?

Reply:
That's existing API.  `launchEngine` is part f our internal API, not available in public headers

Reply:
In other words, we could only ever do this in the embedder implementation - the API to do so is not public.  But it would likely result in bad things.